db/collection.go

```diff
@@ -404,42 +419,86 @@ func (t *Txn) Create(new ...[]byte) ([]core.InstanceID, error) {
 	return results, nil
 }
 
+// Verify verifies updated instances but does not save them.
+func (t *Txn) Verify(updated ...[]byte) error {
+	identity, err := t.token.PubKey()
+	if err != nil {
+		return err
+	}
+	actions, err := t.createSaveActions(identity, updated...)
+	if err != nil {
+		return err
+	}
+
+	events, _, err := t.createEvents(actions)
+	if err != nil {
+		return err
+	}
+	if len(events) == 0 {
+		return nil
+	}
+	for _, e := range events {
+		if err := t.collection.validWrite(identity, e); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 // Save saves an instance changes to be committed when the current transaction commits.
 func (t *Txn) Save(updated ...[]byte) error {
+	identity, err := t.token.PubKey()
+	if err != nil {
+		return err
+	}
+	actions, err := t.createSaveActions(identity, updated...)
+	if err != nil {
+		return err
+	}
+	t.actions = append(t.actions, actions...)
+	return nil
+}
+
+func (t *Txn) createSaveActions(identity thread.PubKey, updated ...[]byte) ([]core.Action, error) {
+	var actions []core.Action
 	for i := range updated {
 		if t.readonly {
-			return ErrReadonlyTx
+			return nil, ErrReadonlyTx
 		}
 
-		item := make([]byte, len(updated[i]))
-		copy(item, updated[i])
+		next := make([]byte, len(updated[i]))
+		copy(next, updated[i])
 
-		if err := t.collection.validInstance(item); err != nil {
-			return err
+		if err := t.collection.validInstance(next); err != nil {
+			return nil, err
 		}
 
-		id, err := getInstanceID(item)
+		id, err := getInstanceID(next)
 		if err != nil {
-			return err
+			return nil, err
 		}
 		key := baseKey.ChildString(t.collection.name).ChildString(id.String())
-		beforeBytes, err := t.collection.db.datastore.Get(key)
+		previous, err := t.collection.db.datastore.Get(key)
 		if err == ds.ErrNotFound {
-			return errCantSaveNonExistentInstance
+			return nil, errCantSaveNonExistentInstance
 		}
 		if err != nil {
-			return err
+			return nil, err
+		}
+		previous, err = t.collection.filterRead(identity, previous)
+		if err != nil {
+			return nil, err
 		}
 
-		t.actions = append(t.actions, core.Action{
+		actions = append(actions, core.Action{
 			Type:           core.Save,
 			InstanceID:     id,
 			CollectionName: t.collection.name,
-			Previous:       beforeBytes,
-			Current:        item,
+			Previous:       previous,
+			Current:        next,
 		})
```

Comment:
I don't feel much confident in having the `go-threads` logic in my head anymore, most prob since I only do big-chunk reviews every some weeks, so a bit hard to follow up. Sorry for, probably, repeated/dumb questions.

So here the patch will be calculated from the filtered/transformed instance regarding the identity. 
Can this break in an unwanted way what other identity with more or less privileges about that same instance when applying this patch?
Saying it differently: are patches generated from different identities with different permissions compatible resulting in wanted changes for all users in the same thread?

Reply:
Yep, well... that's what this change prevents. An identity can only create a patch against its own "view" of the instance, which means that differences are _zeroed_ in the patch.

Reply:
Ah I see. Doing the diff on the original state of the change, not the full one since might be interpreted as wiping hidden attributes

Reply:
Yep, exactly